packages/aws-cdk/lib/api/deploy-stack.ts

```diff
@@ -43,6 +43,15 @@ export async function deployStack(stack: cxapi.SynthesizedStack,
   const cfn = await sdk.cloudFormation(stack.environment, Mode.ForWriting);
   const bodyParameter = await makeBodyParameter(stack, toolkitInfo);
 
+  if (await stackFailedCreating(cfn, deployName)) {
+    debug(`Found existing stack ${deployName} that had previously failed creation. Deleting it before attempting to re-create it.`);
+    await cfn.deleteStack({ StackName: deployName }).promise();
+    const deletedStack = await waitForStack(cfn, deployName, false);
+    if (deletedStack && deletedStack.StackStatus !== 'DELETE_COMPLETE') {
```

Comment:
I don't see how we can ever get `DELETE_COMPLETE` state. To get the status of a deleted stack you need the full ARN (including unique ID), but `deployName` is amost certain to not include that? Or does it?

I assume this works by returning `undefined` as soon as the stack can't be fetched by just its name anymore?